### ISSUE TITLE
request parameter in beforeSave() doesn't contain any stored values sent from android app

### ISSUE THREAD

**Issue by mik-dass** [ParsePlatform/Parse-SDK-JS#39]
This is my android code :

     ParseObject parseObject = new ParseObject("likes");
                    ParseObject parseuser=ParseObject.createWithoutData(ParseUser.class,post.getPosAuthorObjectId());
                    ParseObject parsepost=ParseObject.createWithoutData("posts",post.getObjectId());
                    parsepost.put("postAuthor",parseuser);
                    parseObject.put("ownerId", ParseObject.createWithoutData(ParseUser.class, ParseUser.getCurrentUser().getObjectId()));
                    parseObject.put("postId",parsepost);
                    parseObject.saveInBackground

**I am trying to send some data regarding the user's id along with the post object so as to save the post object into the "likes" class and do changes in the post owner's attributes using masterkey without fetching the id of the post owner in a separate query as it is already stored in my android code .**In my experimental  beforeSave():

    Parse.Cloud.beforeSave("likes",function(request,response){
	    var posts = Parse.Object.extend("posts");
	    var post=new posts();
        var post=request.object.get("postId");
        var user=Parse.Object.extend(Parse.User);
        var user=new Parse.User();
        var  id =post.object.get("postAuthor").id;
        if(id!=null){
        	response.success();
        }
        else{
        	response.error("The id is null");
        }
     });

Basically I am trying to retrieve the user id saved in the postobject and sent to the request parameter .But the output of my experiment  is always null.

**Comment by wangmengyan95** [comment]
Hi @mik-dass, I don't have the context of the `likes` and `post`, but the following is my test code and it works fine
Android side
```java
ParseObject parseuser=ParseObject.createWithoutData(ParseUser.class, "XXXX");
ParseObject object = new ParseObject("likes");
object.put("Owner", parseuser);
object.saveInBackground();
```

Cloudcode side
```js
Parse.Cloud.useMasterKey();
var owner = request.object.get("Owner");
owner.fetch({
  success: function(post) {
    own.set('XX', 'XXX')；
    response.success();
  },
  error: function(error) {
    response.error(error);
  }
});
```
If this still does not work for you, do you get any exception from cloudcode side or android side?

**Comment by mik-dass** [comment]
@wangmengyan95 Thank you for replying.So basically  I should create a new column "owner" to store the id of the post owner. But will that be a good design as it will consist of redundant data ???
What I am trying to do is --

**_User** table

objectId  / attribute

**posts**

objectId / authorId(pointer to _User)

**likes**

objectIId / likerId(pointer to _User who likes the post) / postId(ponter to the post liked)

In my cloud code I try to access the attribute of the user class pointed by the posts class.

So I create a _User object without data ,then created a posts object without data and set the authorId equal to the _User object created earlier.Then I created the likes object using the appropriate values and set the postId to the posts object created earlier.I thought this will help me to reduce a query since I did not wanted to fetch the post owner Id in the cloud code consuming a api call . **But this approach failed and the cloud log showed me that the reponse object received the posts object as a pointer with no data in it** .So am I missing something to be done to make my aprroach work??? Or should I do ahead and use one more api call to fetch the post owner Id???  Or create a new column in the likes class to store objectIds of the post owner??? Please reply.Thank you.

**Comment by wangmengyan95** [comment]
Hi @mik-dass, we do not send the nested ParseObject's full data to cloud code. In order to get the data of the nested ParseObject, you have to do a query or fetch. 
For the design issue, you can also take a look at [here](https://www.parse.com/docs/android/guide#relations), we have a tutorial about how to implement relations in Parse. Hope it helps.

**Comment by mik-dass** [comment]
Ok.Thank you .
